clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -318,4 +321,16 @@ public AttributeList setAttributes(AttributeList list) {
                                       + ".(whitelist/blacklist) is not a valid regular expression");
         }
     }
+
+    @Override
+    public void contextChange(MetricsContext metricsContext) {
+        if (!this.mbeans.isEmpty()) {
+            throw new IllegalStateException("JMX MetricsContext can only be updated before JMX metrics are created");
+        }
+        synchronized (LOCK) {
+            if (this.prefix.isEmpty()) {
+                this.prefix = metricsContext.metadata().get(MetricsContext.NAMESPACE);
```

Comment:
we should check to make sure this does not result in a null prefix.

Reply:
Added a check at the beginning of the method to make sure metrics context has _namespace set